abci/example/kvstore/kvstore.go

```diff
@@ -434,28 +433,34 @@ func (app *Application) execPrepareTx(tx []byte) *types.ExecTxResult {
 }
 
 // substPrepareTx substitutes all the transactions prefixed with 'prepare' in the
-// proposal for transactions with the prefix strips.
+// proposal for transactions with the prefix stripped.
 // It marks all of the original transactions as 'REMOVED' so that
 // Tendermint will remove them from its mempool.
-func (app *Application) substPrepareTx(blockData [][]byte) []*types.TxRecord {
+func (app *Application) substPrepareTx(blockData [][]byte, maxTxBytes int64) []*types.TxRecord {
 	trs := make([]*types.TxRecord, len(blockData))
 	var removed []*types.TxRecord
+	var totalBytes int64
 	for i, tx := range blockData {
+		tx_mod := tx
+		action := types.TxRecord_UNMODIFIED
 		if isPrepareTx(tx) {
 			removed = append(removed, &types.TxRecord{
 				Tx:     tx,
 				Action: types.TxRecord_REMOVED,
 			})
-			trs[i] = &types.TxRecord{
-				Tx:     bytes.TrimPrefix(tx, []byte(PreparePrefix)),
-				Action: types.TxRecord_ADDED,
-			}
-			continue
+			tx_mod = bytes.TrimPrefix(tx, []byte(PreparePrefix))
+			action = types.TxRecord_ADDED
+		}
+		nBytes := int64(len(tx_mod))
+		if totalBytes + nBytes > maxTxBytes {
+			trs = trs[:i]
```

Comment:
Took me a second to understand why this was being done this way. I might recommend instead opting to create a slice with capacity of `len(blockData)` and length 0. I.e., `trs := make([]*types.TxRecord, 0, len(blockData))` and then using `append` to keep track of the length of the list,  `trs := append(trs, &types.TxRecord...)`. This will allocate the same amount as the version you have now, since the slice will be allocated to the capacity, but it doesn't require this re-slicing. This way definitely works, it's just a bit confusing imo.

Reply:
I never really liked "append" and always try to avoid it. But I guess it's due to some (unknown) trauma LOL.
If you find it confusing, it's probably confusing. Bear with me